dimagi/commcare-hq
# Update report filters to prioritize userids if they are passed

## Summary

Link to the ticket - https://dimagi-dev.atlassian.net/browse/SAAS-11658
While searching explicitly for a deactivated user by his userId in the Submission by Form report, the results do not return any user. This was because we have a default filter of `is_active` which was set to `True` in our ESQuery. So deactivated users never made it to final results.

One more thing that I found out was we do not consider the user ids if they are passed with `Active Mobile Worker` or `Deactivated Mobile worker` slugs. So if I have searched for all `Deactivated Mobile workers` with one active mobile worker userid, we would only get deactivated mobile workers and the active mobile worker will be ignored.

The PR focuses on considering the `userIds` if they are explicitly passed in the report slugs.

## Product Description
This change will make the filters more accurate for users and would return the results that they are expecting.
## Safety Assurance
I have tested it out on a local setup with all the combinations possible and I was satisfied with the output that I got.

### Automated test coverage
This change unfortunately does not have good test coverage. But I would be writing tests for this function once we agree on the updated behavior. 

### QA Plan
A small QA exercise would be -
- Have some(2-3) activated mobile workers, deactivated mobile workers, Web users, Demo users, and Admin users for a domain.
- Search in `Submissions By Form` report with different combinations and verify if the results are expected.

### Safety story

This affects a small part of the HQ and I have tried to keep the default behavior of the filters intact unless userIds are passed in the filter.
### Rollback instructions
Reverting the PR should roll back the changes
